lucene/core/src/java/org/apache/lucene/util/bkd/PointWriter.java

```diff
@@ -32,9 +32,12 @@
   /** Add a new point from byte array*/
   void append(byte[] packedValue, int docID) throws IOException;
 
-  /** Add a new point from byteRef */
+  /** Add a new point from byteRef and docId */
```

Comment:
I agree, and already though about removing the method that accept a byte array. I have done it now.
I think for the second comment, I prefer to create a dedicated interface for packedValues and use that instead of bytesRef. The main driver for this method is to prevent deserialising and serialising the docId when copying points between OfflineWriters.

Reply:
I have added a new interface called `PointValue` to abstract out the different formats for on-heap and offline points. The side-effect is that I left the interface method `void append(byte[] packedValue, int docID)` as this is used when we initially populate the BKDWriter.